### DISCUSSION TITLE
Replication order

### DISCUSSION THREAD

**Discussion by lewisl9029** [seaweedfs/seaweedfs#2713] [Q&A] [answered]
Hi there! Just discovered SeaweedFS and it looks super promising for my use case!

One question that I couldn't find an answer to after searching around:

With the [Active-Active replication](https://github.com/chrislusf/seaweedfs/wiki/Filer-Active-Active-cross-cluster-continuous-synchronization) setup, does SeaweedFS have any guarantees around the ordering of which files get replicated first? I.e. does it replicate files with earlier timestamp, then later ones or vice versa?

This would be a super useful property for avoiding having to query all replicas in cases where expected assets are missing due to replication ordering. But I can work around it if there are no guarantees.

Thanks!

**Comment by chrislusf** [comment, marked as answer]
Yes, the messages are ordered.

**Reply by lewisl9029** [reply]
That's awesome to know, thank you for the super quick response!
